### DISCUSSION TITLE
How to fetch from a local copy in a "non-colocated" repo?

### DISCUSSION THREAD

**Discussion by chriskrycho** [jj-vcs/jj#1822] [Q&A] [answered]
I have one copy of `jj` where I simply cloned directly from GitHub using `jj git clone`, but I am having issues related to SSH keys and GH PATs (which I may also try to fix if I can get to the bottom of them!) preventing me from doing `jj git push` successfully there, so I did a normal Git clone and `jj init --git-repo=.` in that clone… but I cannot figure out how to pull from the original clone (which is *not* colocated) into the new clone (which *is*) to enable me to just use `git push` there. So: how do you `fetch` from a non-colocated repo?

**Comment by yuja** [comment, marked as answer]
You can `git fetch` from `.jj/repo/store/git`. It wouldn't preserve JJ metadata (like change id or rewrite history), though.

I think you can also turn the original repo into colocated repo by moving directories and rewriting `.jj/repo/store/git_target` file, but I don't remember the exact steps.

**Reply by chriskrycho** [reply]
I thought that might be, from introspecting the layout; thanks for confirming!

**Reply by landtuna** [reply]
In case anyone runs across this, the steps to colocate after the fact are here: https://github.com/jj-vcs/jj/blob/02497e6a12dddc1f6bf11d8f49e98c74274cf1a6/docs/git-compatibility.md

**Reply by nrbray** [reply]
also here: https://jj-vcs.github.io/jj/latest/git-compatibility/#converting-a-repo-into-a-co-located-repo

this worked for me
